content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -659,6 +659,30 @@ there is no automatic Job restart once the Job status is `type: Failed`.
 That is, the Job termination mechanisms activated with `.spec.activeDeadlineSeconds`
 and `.spec.backoffLimit` result in a permanent Job failure that requires manual intervention to resolve.
 
+### Termination of Job pods
+
+A terminated Job may be either succeeded or failed, and its status is reflected
+by the Job conditions `Complete` or `Failed`, respectively. Prior to v1.31
+these Job conditions are added when the Job termination process is triggered,
+but some pods may still remain running at that point in time.
+
+Since v1.31, when you enable either the `JobManagedBy` or
+`JobPodReplacementPolicy` (enabled by default)
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/), the
+Job controller awaits for termination of all pods before adding a condition
+indicating that the Job is finished (either `Complete` or `Failed`).
+
+Note that, the process of terminating all pods may take a substantial amount
+of time, depending the one Pod's `terminationGracePeriodSeconds` (see
+[Pod termination](#docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)),
+and thus adding the terminal the Job condition, even if the fate of the Job is
+already determined.
+
+If you want to know, or the controller you write, the fate of the Job as soon
```

Comment:
not sure if we need to specify the controller bit here

Reply:
Yeah, me neither, might be just noise. Dropped it.